### PR TITLE
Feature/vr 3327 list commits

### PR DESCRIPTION
**Pull request by evgeniilezhnin**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 1]
```diff
@@ -38,6 +49,54 @@ public Response setCommit(Commit commit, BlobFunction setBlobs, RepositoryFuncti
...
+              session.createQuery(
+                  "From CommitEntity c where c.commit_parent.parent_hash = :parentHash");
+          query.setParameter("parentHash", request.getCommitHead());
+        } else {
+          query =
+              session.createQuery(
+                  "From CommitEntity c WHERE c.commit_parent.child_hash = :childHash AND c.commit_parent.parent_hash = :parentHash");
+          query.setParameter("childHash", request.getCommitBase());
+          query.setParameter("parentHash", request.getCommitHead());
+        }
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: This is wrong. It's not about child and parent. It's about boundaries. We get commits that happened after base (including base) and before head (including head), in time. It doesn't matter what is the association between commits. We will implement a different functionality for the equivalent of `git log`, as that depends on branches, which we currently do not have.

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 2]
```diff
@@ -79,4 +138,63 @@ public Commit getCommit(String commitHash, RepositoryFunction getRepository)
       return commitEntity.toCommitProto();
     }
   }
+
+  @Override
+  public GetCommitBlobRequest.Response getCommitBlob(GetCommitBlobRequest request) {
+    return null;
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: Throw a not implemented instead.

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 3]
```diff
@@ -38,6 +49,54 @@ public Response setCommit(Commit commit, BlobFunction setBlobs, RepositoryFuncti
     }
   }
 
+  @Override
+  public ListCommitsRequest.Response listCommits(ListCommitsRequest request) {
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: This completely ignores the path prefix. Was that intentional?

File: backend/src/main/java/ai/verta/modeldb/versioning/VersioningServiceImpl.java [hunk 4]
```diff
@@ -242,14 +257,37 @@ public void listCommitBlobs(
...
       GetCommitFolderRequest request,
       StreamObserver<GetCommitFolderRequest.Response> responseObserver) {
-    super.getCommitFolder(request, responseObserver);
+    QPSCountResource.inc();
+    try (RequestLatencyResource latencyResource =
+        new RequestLatencyResource(modelDBAuthInterceptor.getMethodName())) {
+      String[] split = request.getPath().split("/");
+      if (split.length < 1) {
+        throw new ModelDBException("empty path specified");
+      }
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/VersioningServiceImpl.java]: Update based on the protos changes.

**Comment by ravishetye** [comment]
for this removing the location_prefix from protos. the v0 of list_prefix will do the following
1. get a list of commits `belonging to the repo` which have timestamp `greater than equal to` the time stamp of commit_base and `less than equal to` the timestamp of commit_head.
1. If either of commit_base or commit_head is not present then we donot apply filtering on **that** end 
**Other than that there is no additional filtering required**

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 5]
```diff
@@ -38,6 +49,54 @@ public Response setCommit(Commit commit, BlobFunction setBlobs, RepositoryFuncti
...
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      Stream<CommitEntity> stream;
+      int pageLimit = request.getPagination().getPageLimit();
+      final int startPosition = (request.getPagination().getPageNumber() - 1) * pageLimit;
```
[the remark this answers is not in the document]
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: outdated

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java [hunk 6]
```diff
@@ -38,6 +49,54 @@ public Response setCommit(Commit commit, BlobFunction setBlobs, RepositoryFuncti
...
+      if (request.getCommitBase().isEmpty() && request.getCommitHead().isEmpty()) {
+        Query<CommitEntity> query =
+            session.createQuery("From " + CommitEntity.class.getSimpleName());
+        final List<CommitEntity> list = query.list();
+        stream = list.stream().filter(c -> c.getChild_commits().isEmpty());
```
[the remark this answers is not in the document]
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: outdated

File: backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java (hunk 1 above)
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java]: outdated

**Pull request merged**